### PR TITLE
ipatest: Test ipa-cert-fix fails when startup directive is missing from CS.cfg

### PR DESCRIPTION
**Pull request by mrizwan93**
This test checks that if 'selftests.container.order.startup' directive
is missing from CS.cfg, ipa-cert-fix fails and throw proper error
message. It also checks that underlying command 'pki-server cert-fix'
should fail to renew the cert.

related: https://pagure.io/freeipa/issue/8721

Signed-off-by: Mohammad Rizwan <myusuf@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
The test looks ok but the comment and commit message I think are too narrowly focused on one failure use-case.

The investigation discovered that we were too optimistic when pki-cert-fix returned an error and continued to try to renew certificates only to end up with an even stranger message. The configuration you're testing matches this report but the underlying fix theoretically would apply to other types of failures.

I guess I'd emphasis more that we are exercising this better checking of general failures than preventing this one specific failure type. This particular failure is just one known way to cause the problem.

**Comment by mrizwan93** [comment]
> The test looks ok but the comment and commit message I think are too narrowly focused on one failure use-case.
> 
> The investigation discovered that we were too optimistic when pki-cert-fix returned an error and continued to try to renew certificates only to end up with an even stranger message. The configuration you're testing matches this report but the underlying fix theoretically would apply to other types of failures.
> 
> I guess I'd emphasis more that we are exercising this better checking of general failures than preventing this one specific failure type. This particular failure is just one known way to cause the problem.

Bug report have expired CA and running ipa-cert-fix. I'll add this scenario.

**Comment by rcritten** [comment]
Better, thanks. Please remove the temp commit.

**Comment by mrizwan93** [comment]
Thanks @rcritten , temp commit removed.

**Comment by rcritten** [comment]
master:

* 16057898af69ee795c9c2871ce5936a49d108e1c ipatest: Test ipa-cert-fix fails when startup directive is missing from CS.cfg

File: ipatests/test_integration/test_ipa_cert_fix.py [hunk 1]
```diff
@@ -192,6 +203,53 @@ def test_ipa_cert_fix_non_ipa(self):
...
+                                 '--ignore-service-failures'])
+
+        result = self.master.run_command(['ipa-cert-fix', '-v'],
+                                         stdin_text='yes\n',
+                                         raiseonerr=False)
+        err_msg1 = "ERROR: 'selftests.container.order.startup'"
+        # check that pki-server cert-fix command fails
+        err_msg2 = ("ERROR: CalledProcessError(Command "
+                    "['pki-server', 'cert-fix'")
+        assert err_msg1 and err_msg2 in result.stderr_text
```
Comment (wladich) [inline ipatests/test_integration/test_ipa_cert_fix.py:235]: This means 
```
assert (err_msg1) and (err_msg2 in result.stderr_text)
```

So it does not check if err_msg1 is conained in output. It only checks that err_msg is not empty.

**Pull request closed without merging**